godot-core/src/builtin/array.rs

```diff
@@ -393,6 +393,14 @@ impl<T: GodotType + FromGodot> Array<T> {
         let variant = unsafe { &*ptr };
         T::from_variant(variant)
     }
+    /// Returns the value at the specified index or `None` if the index is out-of-bounds.
+    pub fn try_get(&self, index: usize) -> Option<T> {
+        let ptr = self.ptr_or_null(index);
+        match ptr.is_null() {
+            true => None,
+            false => Some(T::from_variant(unsafe { &*ptr })),
+        }
```

Comment:
For `bools`, I would use `if/else` rather than `match`.

Reply:
Thanks for the review @Bromeon. I took inspiration from related methods in the same file, would something like this work, style-wise?
```rs
/// Returns the value at the specified index or `None` if the index is out-of-bounds.
    pub fn try_get(&self, index: usize) -> Option<T> {
        let ptr = self.ptr_or_null(index);
        (!ptr.is_null()).then(|| {
            // SAFETY: `!ptr.is_null()` just verified that the index is not out of bounds.
            let variant = unsafe { &*ptr };
            T::from_variant(variant)
        })
    }
```

Reply:
You could even use [`sys::ptr_then`](https://github.com/godot-rust/gdext/blob/master/godot-ffi/src/toolbox.rs#L105):
```rs
    pub fn try_get(&self, index: usize) -> Option<T> {
        let ptr = self.ptr_or_null(index);
        sys::ptr_then(ptr, || {
            // SAFETY: a non-null pointer implies that the index is within bounds.
            let variant = unsafe { &*ptr };
            T::from_variant(variant)
        })
    }
```

Reply:
Much cleaner, however `sys::ptr_then` takes a `*mut ptr`, so we would have to use `self.ptr_mut_or_null()` instead. 
I suppose it's fine to have mutability here, as we're not doing much with it, but looks a bit off for something that fetches a value.

If it's okay, I can amend and force-push with the changes.

Reply:
Yeah, I mean we also don't have to overdo it with the functional style.

`if`/`else` is perfectly fine, and if it's more readable, then let's use that.